src/middleware/timeout/index.ts

```diff
@@ -0,0 +1,61 @@
+import { HTTPException } from '../../http-exception'
+import type { MiddlewareHandler } from '../../types'
+import type { StatusCode } from '../../utils/http-status'
+
+interface TimeoutOptions {
+  message?: string
+  code?: StatusCode
+}
+
+const DEFAULT_ERROR_MESSAGE = 'Gateway Timeout'
+const DEFAULT_ERROR_CODE = 504
+
+interface DurationUnits {
+  [key: string]: number
+}
+
+const parseDuration = (duration: string): number => {
+  const units: DurationUnits = { ms: 1, s: 1000, m: 60000, h: 3600000 }
+  const pattern = /(\d+)(ms|s|m|h)/g
+  let totalMilliseconds = 0
+
+  let match: RegExpExecArray | null
+  while ((match = pattern.exec(duration)) !== null) {
+    const value = parseInt(match[1], 10)
+    const unit = match[2]
+
+    if (!units[unit]) {
+      throw new Error(`Unsupported time unit: ${unit}`)
+    }
+    totalMilliseconds += value * units[unit]
+  }
+
+  return totalMilliseconds
+}
+
+export const timeout = (
+  duration: number | string,
+  options: TimeoutOptions = {}
+): MiddlewareHandler => {
+  const errorMessage = options.message ?? DEFAULT_ERROR_MESSAGE
+  const errorCode = options.code ?? DEFAULT_ERROR_CODE
+  const ms = typeof duration === 'string' ? parseDuration(duration) : duration
+
+  return async (context, next) => {
```

Comment:
For the debugging of users, it would be better to return a named function like other middleware.
https://github.com/honojs/hono/blob/e9e3c8ab65039a8bd2d9f3d961ff8d1fa7a6bdfe/src/middleware/body-limit/index.ts#L50

Reply:
thx! I fixed.